pystac_client/client.py

```diff
@@ -131,7 +131,7 @@ def open(
                 After getting a child collection with, e.g.
                 :meth:`Client.get_collection`, the child items of that collection
                 will still be signed with ``modifier``.
-            request_modifier: A callable that eitehr modifies a `Request` instance or
+            request_modifier: A callable that either modifies a `Request` instance or
```

Comment:
We have it: https://github.com/stac-utils/pystac-client/blob/a5705e0f4bc1ff2cd80b8141310a303a77acbba4/scripts/lint#L20

Neither of these words are in the codespell dictionaries, so they didn't get picked up. I assumed that something was broken, but after some digging I learned that I didn't understand what codespell was. It's just a "common misspellings" checker, not a full spell checker. From their docs:

> It's designed primarily for checking misspelled words in source code, but it can be used with other files as well. It does not check for word membership in a complete dictionary, but instead looks for a set of common misspellings. Therefore it should catch errors like "adn", but it will not catch "adnasdfasdf".

Reply:
> We have it:
> 
> https://github.com/stac-utils/pystac-client/blob/a5705e0f4bc1ff2cd80b8141310a303a77acbba4/scripts/lint#L20
> 
> Neither of these words are in the codespell dictionaries, so they didn't get picked up. I assumed that something was broken, but after some digging I learned that I didn't understand what codespell was. It's just a "common misspellings" checker, not a full spell checker. From their docs:
> 
> > It's designed primarily for checking misspelled words in source code, but it can be used with other files as well. It does not check for word membership in a complete dictionary, but instead looks for a set of common misspellings. Therefore it should catch errors like "adn", but it will not catch "adnasdfasdf".

Yeah, codespell is a better than nothing tool with a lot of gaps.  Thanks for showing me the precommit thing.  That's awesome